src/components/space/functionality/ExportData.js

```diff
@@ -117,6 +121,7 @@ const ExportData = () => {
     setShowLoadingSpinner(true);
     setTimeout(() => {
       requestFullDataset();
+      setShowLoadingSpinner(false);
```

Comment:
Is `requestFullDataset` synchronous, then?

Reply:
no, asynchronous. but when it is called, there are various state changes which are used to update what this component shows. before this PR, we relied on these state updates to stop rendering the loadingSpinner (so there was no need to explicitly set it to false). but, since we can now toggle views and both views use this component, without `setShowLoadingSpinner(false)` the loading spinner would continue to show for the task that is not yet requested, like this:

![ezgif-6-6472802940c1](https://user-images.githubusercontent.com/19311953/99955057-ffc50600-2d83-11eb-85b3-26e0deaeac94.gif)

overall, this component was the trickiest part of the code, since (unlike actions) it is relying on the same context/provider (`TaskDataProvider.js`) for both views. so have another quick glance and let me know if you catch anything that might be off.